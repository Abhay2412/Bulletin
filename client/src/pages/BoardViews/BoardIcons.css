.board-icon-image {
    width: 80%;
    box-shadow: "1px 5px 8px rgba(165, 165, 165)";
}

.board-icon-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
} 


@media screen and (max-width: 600px) {
    .board-icon-image {
        width: 100%;
        box-shadow: "1px 5px 8px rgba(165, 165, 165)";
    }
  }