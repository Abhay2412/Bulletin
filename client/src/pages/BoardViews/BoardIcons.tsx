import { Typography } from "@mui/material";
import usedBoard from "../../imgs/usedBoard.png";
import newBoard from "../../imgs/newBoard.png";
import { Button } from "@mui/material";
import { useNavigate } from "react-router-dom";
import './BoardIcons.css';

interface Props {
  name: string;
  id: number;
}

// Board Images + Names (maximum of 8??)
export default function BoardIcons({ name, id }: Props) {
  const navigate = useNavigate();
  //Create New Board
  if (id === 0) {
    return (
      <div
        className="board-icon-container"
      >
        <Button
          size="small"
          sx={{
            maxWidth: "10vw",
            "&:hover": { backgroundColor: "transparent" },
          }}
          onClick={() => navigate("/create-board")}
        >
          <img
            style={{ boxShadow: "1px 5px 8px rgba(165, 165, 165)" }}
            src={newBoard}
            alt="Add A New Board"
            width="80%"
          ></img>
        </Button>
        <Typography color="primary" variant="h6" fontWeight="bold">
          Create A New Board
        </Typography>
      </div>
    );
  }

  if(name == "") return <></>

  return (
    <div className="board-icon-container">
      <Button size="small" sx={{ maxWidth: "10vw" }}>
        <img
          className="board-icon-image"
          src={usedBoard}
          width="80%"
        ></img>
      </Button>
      <Typography color="primary" variant="h6" fontWeight="bold">
        {name}
      </Typography>
      <Typography variant="body1" color="primary.light" fontWeight="bold">
        Welcome to the {name} board!
      </Typography>
    </div>
  );
}
