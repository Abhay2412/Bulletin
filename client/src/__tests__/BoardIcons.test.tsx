import React from "react";
import BoardIcons from "../pages/BoardViews/BoardIcons";
import { configure, mount, ReactWrapper } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    board_id: "g7HGuvfnGe3nhQgAF8qY",
  }),
}));

describe("Testing <BoardIcons /> Component", () => {
  it("Regular board icon should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <BoardIcons id="1" name="test board" isAdd={false} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Add board icon should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <BoardIcons id="1" name="test board" isAdd={true} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
