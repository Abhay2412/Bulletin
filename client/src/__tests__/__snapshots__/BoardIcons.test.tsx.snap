// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <BoardIcons /> Component Add board icon should render properly 1`] = `
<div
  className="board-icon-container"
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-16xo46n-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="Add A New Board"
      src="newBoard.png"
      style={
        Object {
          "boxShadow": "1px 5px 8px rgba(165, 165, 165)",
        }
      }
      width="80%"
    />
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-1x7nvub-MuiTypography-root"
  >
    Create A New Board
  </h6>
</div>
`;

exports[`Testing <BoardIcons /> Component Regular board icon should render properly 1`] = `
<div
  className="board-icon-container"
>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-4iz0b5-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      className="board-icon-image"
      src="usedBoard.png"
      width="80%"
    />
  </button>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-1x7nvub-MuiTypography-root"
  >
    test board
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1yma4ta-MuiTypography-root"
  >
    Welcome to the 
    test board
     board!
  </p>
</div>
`;
